File: javaAlgo/src/test/java/leetcode/Task59Test.java
package leetcode;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import org.junit.jupiter.api.Test;

class Task59Test {

    private final Task59 task = new Task59();

    @Test
    void example1() {
        assertArrayEquals(
                new int[][]{{1, 2, 3}, {8, 9, 4}, {7, 6, 5}},
                task.generateMatrix(3));
    }

    @Test
    void example2() {
        assertArrayEquals(
                new int[][]{{1}},
                task.generateMatrix(1));
    }
}

File: kotlinAlgo/src/test/kotlin/leetcode/Task59Test.kt
package leetcode

import org.junit.jupiter.api.Assertions.assertArrayEquals
import org.junit.jupiter.api.Test

class Task59Test {
    private val task = Task59()

    @Test
    fun example1() {
        assertArrayEquals(
            arrayOf(intArrayOf(1, 2, 3), intArrayOf(8, 9, 4), intArrayOf(7, 6, 5)),
            task.generateMatrix(3),
        )
    }

    @Test
    fun example2() {
        assertArrayEquals(
            arrayOf(intArrayOf(1)),
            task.generateMatrix(1),
        )
    }
}
